flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -57,6 +60,17 @@ public static void validateConsumerConfiguration(Properties config) {
 				}
 				throw new IllegalArgumentException("Invalid initial position in stream set in config. Valid values are: " + sb.toString());
 			}
+
+			// specified initial timestamp in stream when using AT_TIMESTAMP
+			if (InitialPosition.valueOf(initPosType) == InitialPosition.AT_TIMESTAMP) {
+				if (!config.containsKey(ConsumerConfigConstants.STREAM_INITIAL_TIMESTAMP)) {
+					throw new IllegalArgumentException("Please set value for initial timestamp ('"
+						+ ConsumerConfigConstants.STREAM_INITIAL_TIMESTAMP + "') when using AT_TIMESTAMP initial position.");
+				}
+				validateOptionalDateProperty(config, ConsumerConfigConstants.STREAM_INITIAL_TIMESTAMP,
+					"Invalid value given for initial timestamp for AT_TIMESTAMP initial position in stream. "
+						+ "Must be a valid format: yyyy-MM-dd'T'HH:mm:ss.SSSXXX or non-negative double value. For example, 2016-04-04T19:58:46.480-00:00 or 1459799926.480 .");
```

Comment:
That's really good error messaging for the user :) Like I mentioned above, we need to inform this in the document as well.